src/jitlayers.cpp

```diff
@@ -793,12 +981,38 @@ StringRef JuliaOJIT::getFunctionAtAddress(uint64_t Addr, jl_code_instance_t *cod
 }
 
 
+#ifdef JL_USE_JITLINK
+# if JL_LLVM_VERSION < 140000
+#  warning "JIT debugging (GDB integration) not available on LLVM < 14.0 (for JITLink)"
+void JuliaOJIT::enableJITDebuggingSupport() {}
+# else
+extern "C" orc::shared::CWrapperFunctionResult
+llvm_orc_registerJITLoaderGDBAllocAction(const char *Data, size_t Size);
+
+void JuliaOJIT::enableJITDebuggingSupport()
+{
+    // We do not use GDBJITDebugInfoRegistrationPlugin::Create, as the runtime name
+    // lookup is unnecessarily involved/fragile for our in-process JIT use case
+    // (with the llvm_orc_registerJITLoaderGDBAllocAction symbol being in either
+    // libjulia-codegen or yet another shared library for LLVM depending on the build
+    // flags, etc.).
+    const auto Addr = ExecutorAddr::fromPtr(&llvm_orc_registerJITLoaderGDBAllocAction);
```

Comment:
@lhames this might be of interest to you.

Reply:
Interesting. @dnadlinger Can you describe what made the lookup fragile?

If possible we want to make `::Create` smooth enough to use that it's the natural default, even for in-process setups. (Just because it makes switching between in-process and out-of-process modes trivial)

Reply:
@lhames I'm not actually sure – I didn't want to touch [the existing `JITDylib` setup code](https://github.com/JuliaLang/julia/blob/dacd16f068fb27719b31effbe8929952ee2d5b32/src/jitlayers.cpp#L656-L684), as I'm not very familiar with the history of the Julia codebase, and apparently `orc::DynamicLibrarySearchGenerator::GetForCurrentProcess` wasn't enough to make it work. I didn't check whether that is supposed to work only for symbols in the main image or also for loaded dylibs, but just based on a quick glance, it appeared to be the former. In the "canonical" configuration, Julia links LLVM as a dylib dependency of an internal dylib used by the main process, so rather than trying to figure out what the correct setup is there, it seemed easier to just rely on the host linker to figure this out. Happy to incorporate any suggestions to avoid duplicating the symbol name, though!